src/ir/codegen/gen_iterate_tokens.go

```diff
@@ -13,16 +13,38 @@ func (g *genIterate) Run() error {
 	ctx := g.ctx
 
 	var buf bytes.Buffer
+
+	fmt.Fprintf(&buf, `func handleToken(t *token.Token, cb func(*token.Token) bool) bool {
```

Comment:
Can this function be defined outside of the auto-generated code?
There are examples of this, when normal functions call auto-generated functions.
It's not very convenient to write such code inside strings; and I think there are no strong reasons for this here.

Reply:
This function also needs a comment.
It's not 100% obvious what bool result means.
In particular, it returns true for nil tokens, so it's not "if at least 1 callback returned true".
The behavior should be either simple to get right, or well-docummented.
I'll take a look at its usages, but right now this exact behavior looks odd.

Reply:
Done.